### PR TITLE
fix: Handle non released OpenShift cluster version

### LINKED ISSUE #3174: OpenShiftClient.getVersion returns incorrect version for non released clusters
When working against a non released version where the version string is something like `4.8.0-0.nightly-2021-05-26-071911` the minor and major field are returned as null.

### PR DESCRIPTION
**Pull request by jeffmaury**
## Description

Fix for incorrect version info being returned for non released OpenShift clusters
Fixes #3174

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/OpenShiftVersionInfoTest.java [hunk 1]
```diff
@@ -136,4 +136,95 @@ void testClusterVersioningOpenshift4() {
     assertEquals("4", versionInfo.getMajor());
     assertEquals("2.14", versionInfo.getMinor());
   }
+
+  @Test
+  void testClusterVersioningOpenshift4Unreleased() {
+    server.expect().get().withPath("/openshift/version").andReturn(404, "").always();
```
Comment (rohanKanojia) [inline kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/OpenShiftVersionInfoTest.java:142]: Umm, Is this call required? I think default response from mockserver will be 404 if no expectation is set against this URL.

File: kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/OpenShiftVersionInfoTest.java [hunk 2]
```diff
@@ -136,4 +136,95 @@ void testClusterVersioningOpenshift4() {
...
+      "      }" +
+      "    }" +
+      "  ]," +
+      "  \"kind\": \"ClusterVersionList\"," +
+      "  \"metadata\": {" +
+      "    \"continue\": \"\"," +
+      "    \"resourceVersion\": \"62446\"," +
+      "    \"selfLink\": \"/apis/config.openshift.io/v1/clusterversions\"" +
+      "  }" +
+      "}").once();
```
Comment (rohanKanojia) [inline kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/OpenShiftVersionInfoTest.java:223]: Is it possible to trim parts of response string which are not required for calculating version? Maybe we can move this to `src/test/resources/` and load it from JSON/YAML via `Serialization.jsonMapper().readValue`?

File: kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/OpenShiftVersionInfoTest.java (hunk 1 above)
Comment (jeffmaury) [reply kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/OpenShiftVersionInfoTest.java:142]: Just copied from the existing test above

Comment (rohanKanojia) [reply kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/OpenShiftVersionInfoTest.java:142]: Hmm, I just ran `OpenShiftVersionInfoTest#testClusterVersioningOpenshift4` and it passes even when I delete this. Could you please remove this from here and from the test above too?

**Comment by rohanKanojia** [comment]
@jeffmaury : Could you please review https://github.com/jeffmaury/kubernetes-client/pull/1 whenever you get time?

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**